test/document.js

```diff
@@ -565,7 +565,28 @@ describe('serialize document', function() {
 
     assert.throws(() => {
       firestore.doc('collectionId/documentId').update(obj);
-    }, new RegExp('Argument "dataOrField" is not a valid Document. Input ' + 'object is deeper than 20 levels or contains a cycle.'));
+    }, new RegExp('Argument "dataOrField" is not a valid Document. Input object is deeper than 20 levels or contains a cycle.'));
+  });
+
+  it("doesn't traverse cyclic references", function() {
```

Comment:
Perhaps something like "able to set doc references with cycles" is more  suitable here.